spec/Section 4 -- Composition.md

```diff
@@ -18,4 +18,49 @@ run in sequence to produce the composite execution schema.
 
 ### Post Merge Validation
 
+#### Empty Merged Object Type 
+
+**Error Code**
+
+OBJECT_TYPE_EMPTY
+
+**Formal Specification**
+
+- Let {types} be the set of all object types in the merged schema.
+- For each {type} in {types}:
+  - {IsObjectTypeEmpty(type)} must be false.
+
+IsObjectTypeEmpty(type):
+
+- Let {fields} be a set of all fields of {type} 
+- For each {field} in {fields}:
+  - If {field} is not marked with `@inaccessible`:
+    - return false
+- return true
+
+**Explanatory Text**
+
+For object types defined across multiple source schemas, the merged object type is the superset of all fields defined in these schemas. 
+However, any field marked with `@inaccessible` in any schema is hidden and not included in the merged object type. 
+An object type with no fields, after considering `@inaccessible` markings, is considered empty and invalid.
+
+In the following example, the merged object type `ObjectType1` is valid. 
+
+```graphql
+type ObjectType1 {
+  field1: String
+  field2: Int @inaccessible
+}
+```
+
+This counter-example demonstrates an invalid merged object type. 
+In this case, `ObjectType1` is defined in two source schemas, but all fields are marked as `@inaccessible` in at least one of the schemas, resulting in an empty merged object type:
+
+```graphql counter-example
+type ObjectType1 {
+  field1: String @inaccessible
+  field2: Boolean @inaccessible
```

Comment:
```suggestion
  field2: Int @inaccessible
```
Suggestion, to keep it consistent with `ObjectType1` in the previous example. The types are not relevant to the example, so it helps (IMO) to keep them the same, so that the difference is easier to notice.